api/v1alpha2/cloudprovider_types.go

```diff
@@ -54,6 +54,14 @@ type CPIConfig struct {
 	// CPIProviderConfig contains extra information used to configure the
 	// vSphere cloud provider.
 	ProviderConfig CPIProviderConfig `json:"providerConfig,omitempty"`
+
+	// Route is the vSphere cloud provider's route configuration.
+	// +optional
+	Route CPIRouteConfig `gcfg:"Route,omitempty" json:"route,omitempty"`
```

Comment:
these should use json tags (see examples above) and json fields are camelCase

Reply:
oh I'm realizing now you added both gcfg and json so I think we can just remove the `gcfg` ones.

Reply:
okay ignore me, I see now that this struct is supposed to have gcfg tags for the encoding logic.